src/parser/constructs.rs

```diff
@@ -643,6 +661,22 @@ impl Construct {
         Ok((input, if_else))
     }
 
+    /// Parse return construct. Consumes a return with its potential value
+    ///
+    /// `<return> [ <xxx> ]`
+    pub(crate) fn jk_return(input: &str) -> ParseResult<&str, Return> {
+        let (input, _) = Token::maybe_consume_extra(input)?;
+        let (input, _) = Token::return_tok(input)?;
+        let (input, _) = Token::maybe_consume_extra(input)?;
+
+        let (input, val) = opt(Construct::instruction)(input)?;
+        let (input, _) = opt(Token::semicolon)(input)?;
+
+        let return_inst = Return::new(val);
+
+        Ok((input, return_inst))
```

Comment:
If you want, it's completely fine to create the `Return` on the fly here
```suggestion
        Ok((input, Return::new(val)))
```